.github/workflows/run_tests.yml

```diff
@@ -0,0 +1,48 @@
+name: Run tests
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  run-tests:
+    name: Run tests on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: bash
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - { os: ubuntu-latest }
+          - { os: windows-latest }
+          - { os: macos-latest }
+    steps:
+      - name: Cancel non-latest runs
+        uses: styfle/cancel-workflow-action@0.11.0
+        with:
+          all_but_latest: true
+          access_token: ${{ github.token }}
+
+      - uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install -r requirements-dev.txt -r requirements.txt
+
+      - name: Install package
+        run: |
+          python -m pip install -e .  # must install in editable mode for coverage to find sources
```

Comment:
@rly Should we drop `-e` since we're not running coverage?

Reply:
Sure. It shouldn't make a difference, I think.

Reply:
I added coverage in this PR, so let's keep `-e`